module/Photo/src/Photo/Service/AlbumCover.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Photo\Service;
+
+use Application\Service\AbstractService;
+use Zend\ServiceManager\ServiceManager;
+use Zend\ServiceManager\ServiceManagerAwareInterface;
+use Photo\Model\Album as AlbumModel;
+use Photo\Model\Photo as PhotoModel;
+use Imagick;
+
+/**
+ * Album cover services. Used for (re)generating album covers.
+ * 
+ */
+class AlbumCover extends AbstractService
+{
+
+    public function createCover($album)
+    {
+        $cover = $this->generateCover($album);
+        $tempFileName = sys_get_temp_dir() . '/CoverImage' . rand() . '.png';
+        $cover->writeImage($tempFileName);
+        $newPath = $this->getPhotoService()->generateStoragePath($tempFileName);
+        $config = $this->getConfig();
+        rename($tempFileName, $config['upload_dir'] . '/' . $newPath);
+        return $newPath;
+    }
+
+    protected function generateCover($album)
+    {
+        $config = $this->getConfig();
+        $columns = $config['album_cover']['cols'];
+        $rows = $config['album_cover']['rows'];
+        $count = $columns * $rows;
+        $images = $this->getImages($album, $count);
+        //if there are not enough images to fill the matrix, reduce the rows and columns
+        while (($columns > 1 || $rows > 1) && count($images) < $count) {
```

Comment:
Moet denk ik gewoon

count($images) < $count zijn

Als er geen images zijn maakt het niet zoveel uit dat er geen rows zijn, en als er meer images zijn is een [1 x 1] matrix volgensmij ook gewoon goed